.github/workflows/github-script/generate-wiki-page.js

```diff
@@ -0,0 +1,610 @@
+// @ts-nocheck
+
+const fs = require('fs').promises;
+const { join } = require('path');
+const { load } = require('js-yaml');
+
+async function getWorkspaceList(workspacesDir, core) {
+  try {
+    const entries = await fs.readdir(workspacesDir, { withFileTypes: true });
+    return entries
+      .filter(entry => entry.isDirectory() && !entry.name.startsWith('.'))
+      .map(entry => entry.name)
+      .sort();
+  } catch (error) {
+    core.setFailed(`Error reading workspaces directory ${workspacesDir}: ${error.message}`);
+    return [];
+  }
+}
+
+async function parseSourceJson(workspacePath, core) {
+  const sourceFile = join(workspacePath, 'source.json');
+  try {
+    const content = await fs.readFile(sourceFile, 'utf-8');
+    return JSON.parse(content);
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      core.setFailed(`Error reading ${sourceFile}: ${error.message}`);
+    }
+    return null;
+  }
+}
+
+async function parsePluginsList(workspacePath, core) {
+  const pluginsFile = join(workspacePath, 'plugins-list.yaml');
+  try {
+    const content = await fs.readFile(pluginsFile, 'utf-8');
+    const trimmed = content.trim();
+    if (!trimmed) {
+      return [];
+    }
+
+    const data = load(trimmed);
+    if (typeof data === 'object' && data !== null) {
+      if (Array.isArray(data)) {
+        return data;
+      } else if (typeof data === 'object') {
+        return Object.keys(data);
+      }
+    }
+    return [];
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      core.setFailed(`Error reading ${pluginsFile}: ${error.message}`);
```

Comment:
When we call `setFailed()`, we expect to return directly.
In addition to logging an error, `setFailed()` sets the status of the step as failed. So there is usually not reason to continue the step process, since the workflow job will be considered failed anyway.

But in a number of cases, it seems that you call `setFailed()` in a utility function, but don't check the function result at call site to directly return.

When you don't want to stop and fail the job step, using the `core.error()` or `core.warning()` function might be better then.